cap/modules/schemas/imp.py

```diff
@@ -153,6 +154,28 @@ def _filter_by_read_access(schemas_list):
     return [x for x in schemas_list if ReadSchemaPermission(x).can()]
 
 
+def _filter_by_admin_access(schemas_list):
+    """Return only schemas that user has admin access to."""
```

Comment:
lets have a mapping `key => method`, where we can drop the key and method , to generate the final array. methods should return `true/false` or the a respected array of values

Final roles array can be flattened